std/json.d

```diff
@@ -1680,12 +1680,16 @@ if (isOutputRange!(Out,char))
                 }
                 else
                 {
+                    import std.algorithm.searching : canFind;
                     import std.format : format;
                     // The correct formula for the number of decimal digits needed for lossless round
                     // trips is actually:
                     //     ceil(log(pow(2.0, double.mant_dig - 1)) / log(10.0) + 1) == (double.dig + 2)
                     // Anything less will round off (1 + double.epsilon)
-                    json.put("%.18g".format(val));
+                    auto result = format!"%.18g"(val);
+                    json.put(result);
```

Comment:
Use `formattedWrite` to avoid needless intermediate allocation

Reply:
In the next line I need this intermediate result. I think, I could use `sformat` though to avoid GC allocation.